fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/InviteFragment.kt

```diff
@@ -73,6 +79,35 @@ class InviteFragment : Fragment() {
     LaoActivity.addBackNavigationCallbackToEvents(requireActivity(), viewLifecycleOwner, TAG)
   }
 
+  private fun setupCopyServerButton(binding: InviteFragmentBinding) {
+    val serverTextView = binding.laoPropertiesServerText
+    val copyButton = binding.copyServerButton
+
+    copyButton.setOnClickListener {
+      val text = serverTextView.text.toString()
+      copyTextToClipboard(text)
+      Toast.makeText(requireContext(), R.string.successful_copy, Toast.LENGTH_SHORT).show()
+    }
+  }
+
+  private fun setupCopyIdentifierButton(binding: InviteFragmentBinding) {
+    val identifierTextView = binding.laoPropertiesIdentifierText
+    val copyButton = binding.copyIdentifierButton
+
+    copyButton.setOnClickListener {
+      val text = identifierTextView.text.toString()
+      copyTextToClipboard(text)
+      Toast.makeText(requireContext(), R.string.successful_copy, Toast.LENGTH_SHORT).show()
+    }
+  }
+
+  private fun copyTextToClipboard(token: String) {
+    val clipboard =
+        requireActivity().getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+    val clip = ClipData.newPlainText(token, token)
```

Comment:
Here reading the doc the first param should be the text label that you're copying. So instead of reusing the token as its own label, I'd put an actual label that explains what has been copied (not its content)

Reply:
yes that's what I did first, but then I saw line 100 in [TokenFragment.kt](https://github.com/dedis/popstellar/blob/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/token/TokenFragment.kt) and went to do the same. Maybe I just should have trusted my critical sense :woman_shrugging: 

code was : 
```
  private fun copyTextToClipboard(token: String) {
    val clipboard =
        requireActivity().getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
    val clip = ClipData.newPlainText(token, token)
    clipboard.setPrimaryClip(clip)
  }
```

Reply:
Yes in an utils file would be perfect